app/dao/notifications_dao.py

```diff
@@ -634,7 +634,11 @@ def dao_get_notifications_by_recipient_or_reference(service_id, search_term, not
     if notification_type:
         filters.append(Notification.notification_type == notification_type)
 
-    results = db.session.query(Notification).filter(*filters).order_by(desc(Notification.created_at)).all()
+    results = db.session.query(Notification)\
+        .filter(*filters)\
+        .order_by(desc(Notification.created_at))\
+        .limit(current_app.config['PAGE_SIZE'])\
```

Comment:
Optional: To pass the limit as an argument to the `dao` function and so it's more the view function that decides how many results should be retrieved which is more business logic (unless this a pattern that is already well established or other reasons)?